xplique/attributions/global_sensitivity_analysis/hsic_attribution_method.py

```diff
@@ -0,0 +1,72 @@
+"""
+Hsic Attribution Method explainer
+"""
+
+from ...types import Callable, Union, Optional
+from .gsa_attribution_method import GSABaseAttributionMethod
+from .samplers import Sampler, LatinHypercube
+from .hsic_estimators import (
+    BinaryEstimator,
+    HsicEstimator,
+)
+
+
+class HsicAttributionMethod(GSABaseAttributionMethod):
+    """
+    HSIC Attribution Method.
+    Compute the dependance of each input dimension wrt the output using Hilbert-Schmidt Independance
+    Criterion, a perturbation function on a grid and an adapted sampling as described in
+    the original paper.
+
+    Ref. Novello, Fel, Vigouroux, Making Sense of Dependance: Efficient Black-box Explanations
+    Using Dependence Measure, https://arxiv.org/abs/2206.06219
+
+    Parameters
+    ----------
+    model
+        Model used for computing explanations.
+    grid_size
+        Cut the image in a grid of (grid_size, grid_size) to estimate an indice per cell.
+    nb_design
+        Number of design for the sampler.
+    sampler
+        Sampler used to generate the (quasi-)monte carlo samples, LHS or QMC.
+        For more option, see the sampler module.
+    estimator
+        Estimator used to compute the HSIC score
+    perturbation_function
+        Function to call to apply the perturbation on the input. Can also be string in
+        'inpainting', 'blur'.
+    batch_size
+        Batch size to use for the forwards.
+    """
+
+    def __init__(
+        self,
+        model,
+        grid_size: int = 8,
```

Comment:
A 6 times 6 `grid_size` is recommended in the documentation. Why is the default value different?

Reply:
fixed ;)